server/odc-service/src/main/java/com/oceanbase/odc/service/session/interceptor/DBResourcePermissionInterceptor.java

```diff
@@ -88,13 +90,25 @@ public boolean doPreHandle(@NonNull SqlAsyncExecuteReq request, @NonNull SqlAsyn
         for (Entry<DBSchemaIdentity, Set<SqlType>> entry : identity2Types.entrySet()) {
             DBSchemaIdentity identity = entry.getKey();
             Set<SqlType> sqlTypes = entry.getValue();
+            Set<DatabasePermissionType> permissionTypes = new HashSet<>();
             if (CollectionUtils.isNotEmpty(sqlTypes)) {
-                Set<DatabasePermissionType> permissionTypes = sqlTypes.stream().map(DatabasePermissionType::from)
-                        .filter(Objects::nonNull).collect(Collectors.toSet());
-                if (CollectionUtils.isNotEmpty(permissionTypes)) {
+                if (sqlTypes.contains(SqlType.CREATE)) {
```

Comment:
the logic here is almost same with the logic in `PreCheckRuntimeFlowableTask.java`, can we use an util to do this?

Reply:
getResource2PermissionTypes(Collection<SqlType> sqltypes)

Reply:
done